src/backend/dx11/src/lib.rs

```diff
@@ -2811,29 +2811,110 @@ impl command::CommandBuffer<Backend> for CommandBuffer {
+                Flags: d3d11::D3D11_BUFFER_UAV_FLAG_RAW,
+            };
+
+            let mut uav: *mut d3d11::ID3D11UnorderedAccessView = ptr::null_mut();
+            let hr = device.CreateUnorderedAccessView(
+                buffer.internal.raw as *mut _,
+                &desc,
+                &mut uav as *mut *mut _ as *mut *mut _,
+            );
+
+            if !winerror::SUCCEEDED(hr) {
+                panic!("fill_buffer failed to make UAV failed: 0x{:x}", hr);
+            }
+
+            let uav = ComPtr::from_raw(uav);
 
-        if !winerror::SUCCEEDED(hr) {
-            panic!("fill_buffer failed to make UAV failed: 0x{:x}", hr);
+            self.context
+                .ClearUnorderedAccessViewUint(uav.as_raw(), &[data; 4]);
         }
 
-        self.context
-            .ClearUnorderedAccessViewUint(uav.as_raw(), &[data; 4]);
+        // If there is unaligned portion at the beginning of the sub region
+        // create a new buffer with the fill data to copy into this region
+        if unaligned_num_elements > 0 {
+            debug_assert!(
+                unaligned_num_elements < 4,
+                "The number of unaligned elements is {} but it should be less than 4",
+                unaligned_num_elements
+            );
+
+            let initial_data = [data; 4];
+
+            let initial_data = d3d11::D3D11_SUBRESOURCE_DATA {
+                pSysMem: initial_data.as_ptr() as *const _,
+                SysMemPitch: 0,
+                SysMemSlicePitch: 0,
+            };
+
+            // TODO: consider using a persistent buffer like the working_buffer
+            let desc = d3d11::D3D11_BUFFER_DESC {
+                ByteWidth: core::mem::size_of_val(&initial_data) as _,
+                Usage: d3d11::D3D11_USAGE_DEFAULT,
+                BindFlags: 0,
+                CPUAccessFlags: 0,
+                MiscFlags: 0,
+                StructureByteStride: 0,
+            };
+            let mut temp_buffer = ptr::null_mut::<d3d11::ID3D11Buffer>();
+
+            assert_eq!(
+                winerror::S_OK,
+                device.CreateBuffer(
+                    &desc,
+                    &initial_data,
+                    &mut temp_buffer as *mut *mut _ as *mut *mut _,
+                )
+            );
+
+            let temp_buffer = ComPtr::from_raw(temp_buffer);
+
+            let src_box = d3d11::D3D11_BOX {
+                left: 0,
+                top: 0,
+                front: 0,
+                // TODO: is this supposed to be in bytes?
+                right: (unaligned_num_elements * core::mem::size_of::<u32>() as u32) as _,
+                bottom: 1,
+                back: 1,
+            };
+
+            self.context.CopySubresourceRegion(
+                buffer.internal.raw as _,
+                0,
+                sub.offset as _, // offset in bytes
+                0,
+                0,
+                // ID3D11Buffer cast to ID3D11Resource valid?
```

Comment:
Yes

From: https://docs.microsoft.com/en-us/windows/win32/api/d3d11/nn-d3d11-id3d11buffer

> The ID3D11Buffer interface inherits from ID3D11Resource. 

And because of how COM works, you can freely cast the pointers like this.